src/libYARP_OS/include/yarp/os/FixedSizeBuffersManager-inl.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+
+
+
+
+template <typename T>
+yarp::os::Buffer<T>::Buffer():key(0), dataPtr(nullptr), numOfElements(0){;}
+
+template <typename T>
+yarp::os::Buffer<T>::~Buffer(){;}
+
+
+
+template <typename T>
+T* yarp::os::Buffer<T>::getData() {return dataPtr;}
+
+template <typename T>
+uint32_t yarp::os::Buffer<T>::getSize() {return numOfElements;}
+
+template <typename T>
+T  yarp::os::Buffer<T>::getValue(uint32_t index) throw (std::out_of_range)
+{
+    if(index<numOfElements)
+        return dataPtr[index];
+    else
+        throw std::out_of_range("yarp::os::Buffer::getValue(index): index is out f range");
```

Comment:
Are we sure to introduce exceptions in yarp?
They are a good thing if they are handled through `try{} catch{}` statement, bad otherwise.
In other yarp containers if the index is out of range usually a dummy object is returned ( `return T();`)

Reply:
The behaviour of buffer is quite equal to `std::vector` regarding excpetions: `std::vector` lauches an `std::out_of_range` exception when a `std::vector::at(index)` function is called with a wrong index; in the same way, `yarp::os::Buffer` launches the same exception when set/get functions are invoked with a wrong index.
Both `std::vector` and `yarp::os::Buffer` has operator `[]` that lets user to access an element without any check of index.

Moreover in yarp we use the std::vector and if no `try-catch` statements is used is a developer's choise.
For example I dind't use the `try-catch`  in implementation layer, because when I access to a buffer's element I'm sure that the index is minor of its size.